traffic_ops/ort/atstccfg/remapdotconfig.go

```diff
@@ -0,0 +1,372 @@
+	}
+
+	server := tc.Server{ID: atscfg.InvalidID}
+	if serverID, err := strconv.Atoi(serverNameOrID); err == nil {
+		for _, toServer := range servers {
+			if toServer.ID == serverID {
+				server = toServer
+				break
+			}
+		}
+	} else {
+		serverName := serverNameOrID
+		for _, toServer := range servers {
+			if toServer.HostName == serverName {
+				server = toServer
+				break
+			}
+		}
+	}
+	if server.ID == atscfg.InvalidID {
+		return "", errors.New("server '" + serverNameOrID + " not found in servers")
+	}
+
+	serverName := server.HostName
+
+	cdn, err := GetCDN(cfg, tc.CDNName(server.CDNName))
+	if err != nil {
+		return "", errors.New("getting cdn '" + string(server.CDNName) + "': " + err.Error())
+	}
+
+	serverCDNDomain := cdn.DomainName
+
+	toToolName, toURL, err := GetTOToolNameAndURLFromTO(cfg)
+	if err != nil {
+		return "", errors.New("getting global parameters: " + err.Error())
+	}
+
+	serverProfileParameters, err := GetServerProfileParameters(cfg, server.Profile)
+	if err != nil {
+		return "", errors.New("getting server profile '" + server.Profile + "' parameters: " + err.Error())
+	}
+
+	atsVersionParam := ""
+	for _, param := range serverProfileParameters {
+		if param.ConfigFile != "package" || param.Name != "trafficserver" {
+			continue
+		}
+		atsVersionParam = param.Value
+		break
+	}
+	if atsVersionParam == "" {
+		atsVersionParam = atscfg.DefaultATSVersion
+	}
+
+	atsMajorVer, err := atscfg.GetATSMajorVersionFromATSVersion(atsVersionParam)
+	if err != nil {
+		return "", errors.New("getting ATS major version from version parameter (profile '" + server.Profile + "' configFile 'package' name 'trafficserver'): " + err.Error())
+	}
+
+	deliveryServices, err := GetCDNDeliveryServices(cfg, server.CDNID)
+	if err != nil {
+		return "", errors.New("getting delivery services: " + err.Error())
+	}
+
+	dsIDs := []int{}
+	for _, ds := range deliveryServices {
+		if ds.ID == nil {
+			// TODO log error?
+			continue
+		}
+		dsIDs = append(dsIDs, *ds.ID)
+	}
+
+	isMid := strings.HasPrefix(server.Type, string(tc.CacheTypeMid))
+
+	serverIDs := ([]int)(nil)
+	if !isMid {
+		// mids use all servers, so pass nil=all. Edges only use this current server
+		serverIDs = append(serverIDs, server.ID)
+	}
+
+	dsServers, err := GetDeliveryServiceServers(cfg, dsIDs, serverIDs)
+	if err != nil {
+		return "", errors.New("getting parent.config cachegroup parent server delivery service servers: " + err.Error())
+	}
+
+	dssMap := map[int]map[int]struct{}{} // set of map[dsID][serverID]
+	for _, dss := range dsServers {
+		if dss.Server == nil || dss.DeliveryService == nil {
+			continue // TODO log?
+		}
+		if dssMap[*dss.DeliveryService] == nil {
+			dssMap[*dss.DeliveryService] = map[int]struct{}{}
+		}
+		dssMap[*dss.DeliveryService][*dss.Server] = struct{}{}
+	}
+
+	useInactive := false
+	if !isMid {
+		// mids get inactive DSes, edges don't. This is how it's always behaved, not necessarily how it should.
+		useInactive = true
+	}
```

Comment:
I think this could all just be one line: `useInactive := !isMid`

Reply:
It could. But I'd prefer to leave it within an if-statement for now, because as the preceding comment says, I'm not convinced it should behave this way, and an if-block is easier to modify to something more complex in the future, or to remove outright.

Reply:
If you say so. It'll likely be you doing that, anyway.